qutip/entropy.py

```diff
@@ -244,24 +243,42 @@ def _entropy_relative(rho, sigma, base=e, sparse=False):
     rel_ent : float
         Value of relative entropy.
 
+    References
+    ----------
+
+    See Nielsen & Chuang, "Quantum Computation and Quantum Information",
+    Section 11.3.1, pg. 511 for a detailed explanation of quantum relative
+    entropy.
     """
-    if rho.type != 'oper' or sigma.type != 'oper':
-        raise TypeError("Inputs must be density matrices..")
-    # sigma terms
-    svals = sp_eigs(sigma.data, sigma.isherm, vecs=False, sparse=sparse)
-    snzvals = svals[svals != 0]
+    if not rho.isoper or not sigma.isoper:
+        raise TypeError("Inputs must be density matrices.")
     if base == 2:
-        slogvals = log2(snzvals)
+        log_base = log2
     elif base == e:
-        slogvals = log(snzvals)
+        log_base = log
     else:
         raise ValueError("Base must be 2 or e.")
```

Comment:
Perhaps it would be easier to allow arbitrary log bases?  Just use `log` throughout the function, then at the end return `S / np.log(base)` - pretty sure the maths works out the same.  Not, of course, that anything other than base 2 or base e are useful, but the internal logic is easier that way as well.

Reply:
It's not a bad idea, but all of the other entropy functions follow this `e` or `2` pattern so I'd prefer to keep things consistent for now and consider changing that later.

Reply:
Sounds sensible to me